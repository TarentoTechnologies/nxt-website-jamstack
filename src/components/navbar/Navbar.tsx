import { Link } from "gatsby";
import React from "react";

import { navBarStyles } from "./Navbar.module.css";

interface NavbarProps {
  imgSrc: any;
  imgAltText: any;
  link: any;
}

export const Navbar = ({ imgSrc, imgAltText, link }: NavbarProps) => {
  return (
    <div className={`${navBarStyles}`} data-testid="navbar">
      <Link to={link} rel="canonical">
        <img src={imgSrc} alt={imgAltText} title={imgAltText} />
      </Link>
    </div>
  );
};
