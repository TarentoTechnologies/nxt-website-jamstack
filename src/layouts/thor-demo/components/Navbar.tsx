import { Link } from "gatsby";
import React from "react";
import { useRecoilState } from "recoil";

import ThorLogo from "../../../../static/images/Thor-i-logo.png";
import { themeState } from "../../../states/atoms";
import { thorLogo } from "../../../styles/style-guide/Typography.module.css";
import {
  dark,
  light,
  moonIcon,
  navbarDark,
  navbarLight,
  sliderOff,
  sliderOn,
  sunIcon,
  toggleButton,
} from "../ThorDemo.module.css";

export const Navbar = () => {
  const [theme, setTheme] = useRecoilState(themeState);

  const toggleTheme = () => {
    setTheme(theme === "light" ? "dark" : "light");
  };

  return (
    <div className={`${theme === "dark" ? navbarDark : navbarLight}`}>
      <Link to="/thor">
        <div className={`d-flex align-items-center`}>
          <img src={ThorLogo} alt="Thor Logo" className="pe-2" />
          <span
            className={`${thorLogo} ${theme === "dark" ? "text-white" : ""}`}
          >
            THOR
          </span>
        </div>
      </Link>
      <label className={`${toggleButton} ${theme === "dark" ? dark : light}`}>
        <input
          type="checkbox"
          checked={theme === "dark"}
          onChange={toggleTheme}
        />
        <span className={`${theme === "dark" ? sunIcon : moonIcon}`}></span>
        <span className={`${theme === "dark" ? sliderOn : sliderOff}`}></span>
      </label>
    </div>
  );
};
