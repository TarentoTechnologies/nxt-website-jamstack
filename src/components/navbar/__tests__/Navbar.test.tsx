import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import navbarImg from "../../../../static/images/logo-inner.svg";
import { Navbar } from "../Navbar";

describe("Test if the Navbar component is working as expected", () => {
  it("<Navbar /> matches snapshot", () => {
    const component = render(
      <Navbar
        data-testid="navbar"
        imgSrc={navbarImg}
        imgAltText={"NXT Logo"}
        link={"/"}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<Navbar /> renders without exploding", () => {
    const component = render(
      <Navbar
        data-testid="navbar"
        imgSrc={navbarImg}
        imgAltText={"NXT Logo"}
        link={"/"}
      />
    );
    expect(component.getByTestId("navbar")).toBeInTheDocument();
  });
});
