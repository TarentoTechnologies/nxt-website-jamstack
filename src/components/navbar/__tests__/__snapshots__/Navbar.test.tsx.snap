// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the Navbar component is working as expected <Navbar /> matches snapshot 1`] = `
<div>
  <div
    class="navBarStyles"
    data-testid="navbar"
  >
    <a
      href="/"
      rel="canonical"
    >
      <img
        alt="NXT Logo"
        src="test-file-stub"
        title="NXT Logo"
      />
    </a>
  </div>
</div>
`;
